knora/dsplib/utils/excel_to_json_resources.py

```diff
@@ -16,101 +16,151 @@ def validate_resources_with_schema(json_file: str) -> bool:
         True if the data passed validation, False otherwise
-
     """
     current_dir = os.path.dirname(os.path.realpath(__file__))
-    with open(os.path.join(current_dir, '../schemas/resources-only.json')) as schema:
+    with open(os.path.join(current_dir, "../schemas/resources-only.json")) as schema:
         resources_schema = json.load(schema)
 
     try:
         jsonschema.validate(instance=json_file, schema=resources_schema)
     except jsonschema.exceptions.ValidationError as err:
         print(err)
         return False
-    print('Resource data passed schema validation.')
+    print("Resource data passed schema validation.")
     return True
 
 
-def resources_excel2json(excelfile: str, outfile: str) -> None:
+def _prepare_dataframe(df: pd.DataFrame, required_columns: list[str], location_of_sheet: str) -> pd.DataFrame:
     """
-    Converts properties described in an Excel file into a properties section which can be integrated into a DSP ontology
+    This method takes a pandas DataFrame, strips the column headers from whitespaces and transforms them to lowercase,
+    strips every cell from whitespaces and inserts "" if there is no string in it, and deletes the rows that don't have
+    a value in one of the required cells.
 
     Args:
-        excelfile: path to the Excel file containing the properties
-        outfile: path to the output JSON file containing the properties section for the ontology
+        df: pandas DataFrame
+        required_columns: headers of the columns where a value is required
+        location_of_sheet: for better error messages, provide this information of the caller
 
     Returns:
-        None
+        prepared DataFrame
     """
 
-    # load file
-    wb = load_workbook(excelfile, read_only=True)
-
-    # get overview
-    sheet = wb['classes']
-    resource_list = [c for c in sheet.iter_rows(min_row=2, values_only=True)]
+    # strip column headers and transform to lowercase, so that the script doesn't break when the headers vary a bit
+    new_df = df.rename(columns=lambda x: x.strip().lower())
+    required_columns = [x.strip().lower() for x in required_columns]
+    # strip every cell, and insert "" if there is no valid word in it
+    new_df = new_df.applymap(lambda x: str(x).strip() if pd.notna(x) and re.search(r"[\wäöü]", str(x)) else "")
+    # delete rows that don't have the required columns
+    for req in required_columns:
+        if req not in new_df:
+            raise ValueError(f"{location_of_sheet} requires a column named '{req}'")
+        new_df = new_df[pd.notna(new_df[req])]
+        new_df = new_df[[bool(re.search(r"[\wäöü]", x)) for x in new_df[req]]]
+    if len(new_df) < 1:
+        raise ValueError(f"{location_of_sheet} requires at least one row")
+    return new_df
+
+
+def _row2resource(row: pd.Series, excelfile: str) -> dict[str, Any]:
+    """
+    Method that takes a row from a pandas DataFrame, reads its content, and returns a dict object of the resource
```

Comment:
I don't think one has to write "This method" in docstrings... I do it a lot too, but I think it's more a development-perspective than a usage-perspective. And in many methods it's not done, so it would be more consistent to leave it out here too